### ISSUE TITLE
Check that a capsule does not include an @Local signature field

### ISSUE THREAD

**Issue by dwtj** [paninij/paninij#152]
A signature cannot be instantiated. Therefore a capsule or signature cannot include an `@Local` signature. For example, if `Stream` is a signature, then this is a snippet of an invalid capsule:

```
...
import org.paninij.ex.helloworld.Stream;

@Capsule
public class LocalSignatureTemplate
{
    @Local Stream stream;
}
```

We need to implement a check that detects this case.

Note that as of commit `8ecb85c` on my fork, this example is encoded as a compile test. Of course, it is currently failing because no check is detecting this problem.

Edit: The name of the current compile test for this is `org.paninij.proc.check.capsule.fields.LocalSignatureTemplate`

**Comment by dwtj** [comment]
Note that correct resolution of this issue may call for the correct resolution of #143.

**Comment by dwtj** [comment]
Wait, wasn't this issue resolved with PR #127 with the implementation of

    org.paninij.proc.check.template.NoLocalSignatureCheck

Sorry that I'm confused on this. Is this check not working properly because of the problem described in #143.

**Comment by jlmaddox** [comment]
#143 is relevant in this case, and is why the test isn't passing. The check appears to work in, say, an Eclipse project once it generates the signature. However, it will not trigger on the first compile attempt since that generated signature class doesn't exist, hence the test error you're getting.

One solution we discussed for this is, on the second round of the processor, re-running specific checks over the files provided from the first round since the generated classes are provided in the second round. Then we would check the TypeMirrors (which should no longer be Error types) and see what annotations they have.

An alternative to that is we could do some sort of prediction saying "this type's name indicates that it's going to be a signature once things are generated".

**Comment by dwtj** [comment]
Ok. Now I remember this issue better. Thanks for the reminder. Both of the solutions which you mentioned might be appropriate ways of addressing #143.

**Comment by dwtj** [comment]
Closed by commit 33e4ff0.
